Mathlib/Algebra/Module/CharacterModule.lean

```diff
@@ -33,31 +34,32 @@ an `R`-lineara map `l : M ⟶ N` induces an `R`-linear map `l⋆ : f ↦ l ∘ f
 
 open CategoryTheory
 
-universe u w w'
-variable (R : Type u) [CommRing R]
-variable (M : Type w) [AddCommGroup M] [Module R M]
-variable (D : Type w) [AddCommGroup D] [Module R D]
-variable (N : Type w') [AddCommGroup N] [Module R N]
+universe uR uM uN uD
+variable (R : Type uR) [CommRing R]
+variable (M : Type (max uR uM)) [AddCommGroup M] [Module R M]
+variable (N : Type (max uR uN)) [AddCommGroup N] [Module R N]
+variable (D : Type (max uR uD)) [AddCommGroup D] [Module R D]
+variable [injective_dual : Injective <| ModuleCat.of R D]
 
 /--
 If `M` is an abelian group, its character module is defined to be the `Hom_ℤ(M, ℚ/ℤ)`
 -/
-def CharacterModule : Type w := M →ₗ[ℤ] (ULift.{w} <| AddCircle (1 : ℚ))
+def CharacterModule : Type (max uR uM uD) := M →ₗ[R] D
```

Comment:
Why do you need `uR` here? And why do you need to write `CharacterModule.{uR, uM, uD}` everywhere?